src/lib/EditorToolbar.js

```diff
@@ -98,6 +100,18 @@ export default class EditorToolbar extends Component {
 //         {this._renderBlockTypeDropdown()}
 //         {this._renderUndoRedo()}
         {buttonsGroups}
+        {customControls && customControls.map((f) => {
+          switch (typeof f) {
+            case 'function':
+              return f(
+                (key, value) => this.setState({['customControl' + key]: value}),
+                (key) => this.state['customControl' + key],
```

Comment:
This is an interesting approach. I'd prefer if we don't use string concatenation, but instead make an object for customControls that can be clearly typed.
```js
(key, value) => this.setState(
  ({customControls}) => ({customControls: {...customControls, [key]: value}})
),
(key) => this.state.customControls[key]
```

Then we can type state like so:
```js
type State = {
  showLinkInput: boolean;
  showImageInput: boolean;
  customControls: {[key: string]: string}
};
```